realm/src/main/java/io/realm/Realm.java

```diff
@@ -1873,4 +1876,44 @@ public static Object getDefaultModule() {
         void execute(Realm realm);
     }
 
+    /**
+     * Encapsulates an async {@link RealmQuery}.
+     * <p>
+     * This will run the {@link RealmQuery} on a worker thread, then invoke this callback on the caller thread
+     */
+    public interface QueryCallback<E extends RealmObject> {
+        void onSuccess (RealmResults<E>  results);
+        void onError (Throwable t);
```

Comment:
Throwable -> Exception according to how the rest of the code work. Also we probably want to crash on all Errors anyway.